R/zchunk_L2322.Fert.R

```diff
@@ -76,272 +76,456 @@ module_energy_L2322.Fert_DISABLED <- function(command, ...) {
     L142.ag_Fert_NetExp_MtN_R_Y <- get_data(all_data, "L142.ag_Fert_NetExp_MtN_R_Y")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # 0. Give binding for variable names used in pipeline
+
+    LEVEL2_DATA_NAMES <- year.fillout <- to.value <- technology <- year <-
+      share.weight <- supplysector <- subsector <- coefficient <- minicam.energy.input <-
+      NEcost_75USDkgN <- input.cost <- remove.fraction <- half.life <- median.shutdown.point <-
+      value <- calOutputValue <- sector <- fuel <- subs.share.weight <- region <- fixedOutput <- . <- NULL
+
     # ===================================================
+    # 1. Perform computations
+    # Create tables to delete technologies and subsectors in regions where heat is not modeled as a fuel
+    # 1a. Supplysector information
+    # L2322.Supplysector_Fert: Supply sector information for fertilizer sector
+    A322.sector %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["Supplysector"]], GCAM_region_names) ->
+      L2322.Supplysector_Fert
+
+    # L2322.FinalEnergyKeyword_Fert: Supply sector keywords for fertilizer sector
+    A322.sector %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["FinalEnergyKeyword"]], GCAM_region_names) %>%
+      na.omit ->
+      L2322.FinalEnergyKeyword_Fert
+
+    # 2b. Subsector information
+    # L2322.SubsectorLogit_Fert: Subsector logit exponents of fertilizer sector
+    A322.subsector_logit %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["SubsectorLogit"]], GCAM_region_names) ->
+      L2322.SubsectorLogit_Fert
+
+    # L2322.SubsectorShrwt_Fert and L2322.SubsectorShrwtFllt_Fert: Subsector shareweights of fertilizer sector
+    tibble(x = NA) %>%
+      add_units("None") %>%
+      add_comments("Not generated") %>%
+      add_flags(FLAG_NO_TEST) ->
+      L2322.SubsectorShrwt_Fert
```

Comment:
Why are we even outputting this if it's never going to be generated?

Reply:
If users read in data to the "year" column of A322.subsector_shrwt.csv, these tables will need to be written out. So, agreed that there's no point in writing out these tables as coded, because they'll be non-existent no matter what. They should just be tables whose nrow = 0 right now, but that would exist if years were read in to the year column. To do this, I'd suggest just temporarily switching the data in the "year.fillout" column to the "year" column.

Reply:
Should we write the code so it will work whether the "year" column exists or not?